nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/CommentUtils.java

```diff
@@ -133,13 +135,22 @@ public static String getExpectedMailContent(DocumentModel commentDocModel, Docum
             Event event, String commentEventType) {
         URL url = CommentUtils.class.getResource("/templates/commentNotificationMail.txt");
         try {
-            String content = Files.readString(Paths.get(url.toURI()));
             var model = Map.of("COMMENT_AUTHOR", commentDocModel.getPropertyValue(COMMENT_AUTHOR), //
                     "COMMENT_ACTION", COMMENT_ADDED.equals(commentEventType) ? "added" : "updated", //
                     "COMMENTED_DOCUMENT", commentedDocModel.getName(), //
                     "COMMENT_DATE", EVENT_DATE_FORMAT.format(Date.from(Instant.ofEpochMilli(event.getTime()))), //
                     "COMMENT_TEXT", commentDocModel.getPropertyValue(COMMENT_TEXT), //
                     "COMMENT_SUBSCRIPTION_NAME", COMMENT_ADDED.equals(commentEventType) ? "New" : "Updated");
+            DocumentModel parentComment = (DocumentModel) event.getContext().getProperties().get(PARENT_COMMENT);
+            if (COMMENT_DOC_TYPE.equals(parentComment.getType())) {
```

Comment:
No risk of NPE on `parentComment` here?

Reply:
parentComment is not the best name for it. It is the source doc of the event so parentComment can actually be the commented top level document. It is always available: https://github.com/nuxeo/nuxeo/blob/3e2515ba8451b945c0e3262aa08aa59e760a30f7/nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AbstractCommentManager.java#L122
TOP_LEVEL_DOCUMENT is always the top level, when it differs from PARENT_COMMENT then we know the new comment is a reply.